src/section/section_map.rs

```diff
@@ -167,66 +134,31 @@ impl SectionMap {
     }
 
     pub fn add_neighbour(&mut self, elders_info: EldersInfo) {
-        let prefix = elders_info.prefix;
         let parent_prefix = elders_info.prefix.popped();
         let sibling_prefix = elders_info.prefix.sibling();
-        let new_elders_info_version = elders_info.version;
-
-        if let Some(old_elders_info) = self.neighbours.insert(prefix, elders_info) {
-            if old_elders_info.version > new_elders_info_version {
-                log_or_panic!(
-                    log::Level::Error,
-                    "Ejected newer neighbour info {:?}",
-                    old_elders_info
-                );
-            }
-        }
+
+        let _ = self.neighbours.insert(elders_info.prefix, elders_info);
 
         // If we just split an existing neighbour and we also need its sibling,
-        // add the sibling prefix with the parent prefix sigs.
-        if let Some(sinfo) = self
-            .neighbours
-            .get(&parent_prefix)
-            .filter(|pinfo| {
-                pinfo.version < new_elders_info_version
-                    && self.our().prefix.is_neighbour(&sibling_prefix)
-                    && !self.neighbours.contains_key(&sibling_prefix)
-            })
-            .cloned()
-        {
-            let _ = self.neighbours.insert(sibling_prefix, sinfo);
+        // add the sibling prefix with the parent info.
+        if let Some(parent_info) = self.neighbours.remove(&parent_prefix) {
+            if self.our().prefix.is_neighbour(&sibling_prefix)
+                && !self.neighbours.contains_key(&sibling_prefix)
```

Comment:
why not insert directly? the old parent_info shall not be replaced?

Reply:
Not sure what you mean by directly?

Reply:
I mean:
```rust
if self.our().prefix.is_neighbour(&sibling_prefix) {
  let _ = self.neighbours.contains_key(sibling_prefix, parent_info);
}
```
unless you mean when we already have that sibling_prefix in `self.neighbours`, then the info shall not be updated (or no need to update) ?

Reply:
Yes, if `sibling_prefix` already exists, then we should not overwrite it with `parent_info`.

Reply:
Ah, ok then.